p2p/src/peer_manager/mod.rs

```diff
@@ -1588,6 +1587,27 @@ where
 pub trait Observer {
     fn on_peer_ban_score_adjustment(&mut self, address: SocketAddress, new_score: u32);
     fn on_peer_ban(&mut self, address: BannableAddress);
+    // This will be called at the end of "heartbeat" function.
+    fn on_heartbeat(&mut self);
+    // This will be called for both incoming and outgoing connections.
+    fn on_connection_acccepted(&mut self, address: SocketAddress);
```

Comment:
FYI: I ended up not needing these functions in the final implementation of the tests; but I decided to keep them (and the corresponding `PeerManagerNotification` variants), because they may become useful in the future.